drivers/gpio/gpio_mcp23x17.c

```diff
@@ -44,6 +48,24 @@ static int read_port_regs(const struct device *dev, uint8_t reg,
 	struct mcp23x17_drv_data *const drv_data =
 		(struct mcp23x17_drv_data *const)dev->data;
 	int ret;
+
+#if MCP23X17_BUS_I2C
```

Comment:
Refactor this like I've shown for write, so that the `sys_le16_to_cpu(port_data)` code block at the end of the function is used by both bus implementations.

Reply:
Endianness should be fixed now.
Need to do some testing on HW to verify changes.